src/cmd/keys.js

```diff
@@ -501,16 +502,30 @@ module.exports = class KeysCommand {
 	}
 
 	async _dfuWrite(device, buffer, { altSetting, startAddr, leave, noErase }) {
-		await device.writeOverDfu(buffer, { altSetting, startAddr, leave, noErase });
+		try {
+			await device.writeOverDfu(buffer, { altSetting, startAddr, leave, noErase });
+		} catch (err) {
+			if (err instanceof DeviceProtectionError) {
+				throw new Error('Operation could not be completed due to device protection.');
+			}
+			throw new VError(ensureError(err), 'Writing over DFU failed');
```

Comment:
This is a change in behavior. Previously it would have just thrown the error directly (i.e. `throw err`). Does this change fix a specific thing you noticed during testing?